src/java.desktop/share/classes/javax/swing/plaf/synth/SynthLookAndFeel.java

```diff
@@ -615,14 +615,28 @@ public void load(InputStream input, Class<?> resourceBase) throws
      * <code>new URL(synthFile, path)</code>. Refer to
      * <a href="doc-files/synthFileFormat.html">Synth File Format</a> for more
      * information.
+     * <p>
+     * Whilst this API may be safe for loading local resources that are
+     * delivered with a {@code LookAndFeel} or application, and so have an
+     * equal level of trust with application code, using it to load from
+     * remote resources, particularly any which may have a lower level of
+     * trust, is strongly discouraged.
+     * The alternative mechanisms to load styles from an {@code InputStream}
+     * {@link load(InputStream, Class)}
+     * using a resource co-located with the application or by providing a
+     * {@code SynthStyleFactory} to {@link setStyleFactory} are preferred.
+     * Consequently this method is deprecated and will be removed in a future release.
      *
      * @param url the <code>URL</code> to load the set of
      *     <code>SynthStyle</code> from
      * @throws ParseException if there is an error in parsing
      * @throws IllegalArgumentException if synthSet is <code>null</code>
      * @throws IOException if synthSet cannot be opened as an <code>InputStream</code>
      * @since 1.6
+     * @deprecated Use {@link load(InputStream, Class)} or
+     * {@link setStyleFactory} instead
```

Comment:
The link to setStyleFactory was used like "{@link #setStyleFactory}" earlier
https://github.com/openjdk/jdk/blob/master/src/java.desktop/share/classes/javax/swing/plaf/synth/SynthLookAndFeel.java#L77

shouldn't we keep it same?

Reply:
where was this ? And why ? This works just fine.

Reply:
I guess there's different ways but AFAIS, java.desktop javadocs use 
"@link xxx to link to a class
and "@link #xxx" to link to a field, constant or method in a class

for example, in SYnthLookAndFeel class only, we have this 
_{@link #load} method, or provide your own {@link SynthStyleFactory}
  to {@link #setStyleFactory}_

ANother example 
```
java/awt/Color.java:     * Throws an {@link IllegalArgumentException} if the value is out of
java/awt/Color.java:     * {@link ColorModel}.
java/awt/Color.java:     * format used by the method {@link #getRGB() getRGB}.
```

and here setStyleFactory is a method, so I asked...